pkg/server/server.go

```diff
@@ -391,156 +550,14 @@ func (s *Server) Run(ctx context.Context) error {
 	// TODO: split apart everything after this line, into their own commands, optional launched in this process
 
 	controllerConfig := rest.CopyConfig(s.IdentityConfig)
+	controllerConfig.Timeout = time.Second * 30
 
-	if err := s.installOpenAPIv3Controller(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installKubeNamespaceController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installClusterRoleAggregationController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installKubeServiceAccountController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installKubeServiceAccountTokenController(ctx, controllerConfig); err != nil {
-		return err
-	}
-
-	if err := s.installRootCAConfigMapController(ctx, s.Apis.GenericAPIServer.LoopbackClientConfig); err != nil {
-		return err
-	}
-
-	if err := s.installApiExportIdentityController(ctx, controllerConfig); err != nil {
+	// Adding the bootup time controllers and informers initialization
+	s.controllerLock.Lock()
+	defer s.controllerLock.Unlock()
```

Comment:
Isn't `(s *Server) Run` blocking due to the ` s.MiniAggregator.GenericAPIServer.PrepareRun().Run(ctx.Done())` call at the end? How would the leader election be able to take this lock when it's already taken by the main server method for its whole lifetime?

Reply:
Moved the unlock after installController to not block startControllers